### PR TITLE
Chapter 6 - Concurrency: Fix process method receiver

### PR DESCRIPTION
**Pull request by ep4sh**
It looks like we need to use *Worker type in the receiver as you did in the code below - line 1792. 
As per your suggestion to choose ref types (to decrease memory consumption) and keep a uniform code style.
I am new in Go, please correct me if I wrong ^___^

### CODE REVIEW

**Comment by karlseguin** [comment]
Not really.

I wish I had gone into more detail on this subject.

There's two considerations: mutability and performance. 

If you need the data to be mutable (if you need changes in a function to be seen by the caller), then it _has_ to be a reference. And if you want to make sure the function can't change the data, it needs to be a copy (or not a reference). So that's a simple choice.

In this case, the data isn't mutated (the `id` field is just read). And it actually makes sense this way. The function probably shouldn't be allowed to change `id`.

But if we didn't really care one way or another about mutability (which we don't really care in this case) it still isn't as simple as: references are smaller thus faster. First, a reference is an indirection, so it's slower to lookup `(&worker).id` vs `(worker).id`. In the first case you have to get the address of the heap from the stack, then read the value on the heap. In the second case, the id is directly on the stack.

Also, specific to this case, since `Worker` only has a single int, it's the same size as the reference/pointer (both are 64bits)...so in this specific case, a reference is going to be worse because: it's the same size AND it adds a level of indirection.

To recap. The first thing we _have_ to do is decide if changes made inside the function should be visible to the caller. If the changes should be visible and we expect it to change, we have to pass a reference. If the change should not be visible under any circumstance, we have to pass a value/copy. If it shouldn't change, but we don't really care (which is probably the most common case), such as calling an internal function that we know/trust to not change the object, and we want to make a performance-oriented choice, then broadly speaking, it depends on  how big the object is.
